business/istio_status.go

```diff
@@ -223,3 +209,39 @@ func GetDeploymentStatus(d apps_v1.Deployment) string {
 	}
 	return status
 }
+
+func (iss *IstioStatusService) getAddonComponentStatus() IstioComponentStatus {
+	extServices := config.Get().ExternalServices
+	ics := IstioComponentStatus{}
+
+	ics.merge(getAddonStatus("prometheus", true, extServices.Prometheus.URL, true))
+	ics.merge(getAddonStatus("grafana", extServices.Grafana.Enabled, extServices.Grafana.InClusterURL, extServices.Grafana.CoreComponent))
+	ics.merge(getAddonStatus("jaeger", extServices.Tracing.Enabled, extServices.Tracing.InClusterURL, extServices.Tracing.CoreComponent))
+
+	return ics
+}
+
+func getAddonStatus(name string, enabled bool, url string, isCore bool) IstioComponentStatus {
+	ics := make([]ComponentStatus, 0)
+
+	// When the addOn is disabled, don't perform any check
+	if !enabled {
+		return ics
+	}
+
+	// Call the addOn service endpoint to find out whether is reachable or not
+	resp, err := http.Get(url)
```

Comment:
Please, use a timeout :-)
cf https://github.com/kiali/kiali/issues/3406

Also, I believe you should use the authentication config, else you could get 403 error.
Two good reasons to use our `httputil.HttpGet` function

Reply:
Really good point! Thanks for catching it.